lib/t/struct/acts_as_comparable.rb

```diff
@@ -7,24 +7,16 @@ module ActsAsComparable
       extend T::Sig
       include ::Comparable
 
-      LESS_THAN_OTHER = -1
-      EQUAL = 0
+      NOT_COMPARABLE = nil
 
       sig { params(other: Object).returns(Integer) }
       def <=>(other)
-        result = EQUAL
-        return LESS_THAN_OTHER if other.class != T.unsafe(self).class
+        return NOT_COMPARABLE if other.class != T.unsafe(self).class
 
-        T.unsafe(self).class.decorator.props.keys.map do |attribute_key|
+        T.unsafe(self).class.decorator.props.keys.each do |attribute_key|
           compare_result = T.unsafe(self).send(attribute_key) <=> other.send(attribute_key)
-          result = if compare_result.nil?
-                    LESS_THAN_OTHER
-                   else
-                     T.cast(compare_result, Integer)
-                   end
-          break if result != EQUAL
+          return T.cast(compare_result, Integer) if compare_result != EQUAL
         end
-        result
```

Comment:
And here if we've made it past the loop through the props we'd probably want to return `0`

Reply:
Ah yes, good catch. I guess we should keep the `EQUAL` constant and return that.